watchtower/wtclient/backup_task.go

```diff
@@ -0,0 +1,344 @@
+package wtclient
+
+import (
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/txsort"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/watchtower/blob"
+	"github.com/lightningnetwork/lnd/watchtower/wtdb"
+)
+
+// backupTask is an internal struct for computing the justice transaction for a
+// particular revoked state. A backupTask functions as a scratch pad for storing
+// computing values of the transaction itself, such as the final split in
+// balance if the justice transaction will give a reward to the tower. The
+// backup task has three primary phases:
+//  1. Init: Determines which inputs from the breach transaction will be spent,
+//     and the total amount contained in the inputs.
+//  2. Bind: Asserts that the revoked state is eligible under a given session's
+//     parameters. Certain states may be ineligible due to fee rates, too little
+//     input amount, etc.. Backup of these states can be deferred to a later
+//     time or session with more favorable parameters. If the session is bound
+//     successfully, the final session-dependent values to the justice
+//     transaction are solidified.
+//  3. Send: Once the task is bound, it will be queued to send to a specific
+//     tower corresponding to the session in which it was bound. The justice
+//     transaction will be assembled by examining the parameters left as a
+//     result of the binding. After the justice transaction is signed, the
+//     necessary components are stripped out and encrypted before being sent to
+//     the tower in a StateUpdate.
+type backupTask struct {
+	chanID       lnwire.ChannelID
+	commitHeight uint64
+	breachInfo   *lnwallet.BreachRetribution
+
+	// state-dependent variables
+
+	toLocalInput  input.Input
+	toRemoteInput input.Input
+	totalAmt      btcutil.Amount
+
+	// session-dependent variables
+
+	blobType      blob.Type
+	sweepAmt      int64
+	rewardAmt     int64
+	rewardAddress []byte
+}
+
+// newBackupTask initializes a new backupTask and populates all state-dependent
+// variables.
+func newBackupTask(chanID *lnwire.ChannelID,
+	breachInfo *lnwallet.BreachRetribution) *backupTask {
+
+	// Parse the non-dust outputs from the breach transaction,
+	// simultaneously computing the total amount contained in the inputs
+	// present. We can't compute the exact output values at this time
+	// since the task has not been assigned to a session, at which point
+	// parameters such as fee rate, number of outputs, and reward rate will
+	// be finalized.
+	var (
+		totalAmt      int64
+		toLocalInput  input.Input
+		toRemoteInput input.Input
+	)
+
+	// If the to-local output is non-dust, add it to the set of candidate
```

Comment:
Comments and variable name here doesn't match. Comment references `to-local`, but the `RemoteOutputSignDesc` is checked. Similar applies to below.